src/include/duckdb/function/aggregate_function.hpp

```diff
@@ -92,7 +93,7 @@ class AggregateFunction : public BaseScalarFunction {
 	                         LogicalType(LogicalTypeId::INVALID), null_handling),
 	      state_size(state_size), initialize(initialize), update(update), combine(combine), finalize(finalize),
 	      simple_update(simple_update), window(window), bind(bind), destructor(destructor), statistics(statistics),
-	      serialize(serialize), deserialize(deserialize) {
+	      serialize(serialize), deserialize(deserialize), order_dependent(AggregateOrderDependent::ORDER_DEPENDENT) {
```

Comment:
Being order dependent is very unusual - I think it makes developing other aggregates simpler to make that the default.

Reply:
I thought about that as well, I opted for not doing so for two reasons:

* There are many aggregates that are "surprisingly" order dependent, like the floating point sum that you mentioned, or anything with floating point instabilities. In that case we might still want to allow users to add an `ORDER BY` clause. 
* It is quite tough to automatically test whether or not an aggregate is order dependent - so it is not easy to notify the developer that they have set this property incorrectly. I imagine this will lead to subtle bugs when writing new aggregates. This is essentially opting to perform the ORDER BY unless the developer is certain the aggregate is not order dependent.